packages/core/echo/echo-schema/src/getter.ts

```diff
@@ -48,17 +50,21 @@ export const getMeta = <T extends {}>(obj: T): ObjectMeta => {
 
 export const isDeleted = <T extends {}>(obj: T): boolean => {
   const proxyHandlerSlot = getProxyHandlerSlot(obj);
-  return proxyHandlerSlot.handler?.isObjectDeleted(obj) ?? false;
+  return proxyHandlerSlot.handler?.isDeleted(obj) ?? false;
 };
 
+// TODO(burdon): Replace most uses with getTypename (and rename itemId property).
 export const getType = <T extends {}>(obj: T | undefined): Reference | undefined => getTypeReference(getSchema(obj));
 
+export const getTypename = <T extends {}>(obj: T | undefined): string | undefined =>
```

Comment:
not against this, but why not stick with `getType(obj).name`?

Reply:
i didn't want to rename `itemId => name` since i don't know what that would break.
also i think having an accessor for typename isn't a bad thing.
can you take another pass anyway (soon) to cleanup all of this.